src/timesync/timesyncd.c

```diff
@@ -94,7 +96,7 @@ static int load_clock_timestamp(uid_t uid, gid_t gid) {
 
 int main(int argc, char *argv[]) {
         _cleanup_(manager_freep) Manager *m = NULL;
-        const char *user = "systemd-timesync";
+        const char *user = "systemd-timesyncd";
```

Comment:
hmm, so, I don't like this part. I think we should stick to the name `systemd-timesync` for this user, already for compat reasons. DynamicUser=1 after all will pick up a static user if one exists by the indicated name, and it will exist on all upgraded systems. Or to say this differently: if a system already has a user "systemd-timesync" and now gains a dynamic "systemd-timesyncd" user, too, then this would be needlessly confusing.

Also, I think it's actually slightly prettier if for unix daemons whose name ends in -d the user doesn't end in a -d...

Reply:
I posted a patch in #7005 now, that restores the `User=` line also for systemd-journal-gatewayd.service, following the same logic. I hope that makes sense.